cmake/OpenCVDetectCUDA.cmake

```diff
@@ -106,7 +106,18 @@ if(CUDA_FOUND)
   if(OPENCV_CUDA_DETECTION_NVCC_FLAGS MATCHES "-ccbin")
     # already specified by user
   elseif(CUDA_HOST_COMPILER AND EXISTS "${CUDA_HOST_COMPILER}")
-    LIST(APPEND OPENCV_CUDA_DETECTION_NVCC_FLAGS -ccbin "${CUDA_HOST_COMPILER}")
+    get_filename_component(c_compiler_realpath "${CMAKE_C_COMPILER}" REALPATH)
+    # C compiler doesn't work with --run option, forcing C++ compiler instead
+    if(CUDA_HOST_COMPILER STREQUAL ${c_compiler_realpath} OR CUDA_HOST_COMPILER STREQUAL ${CMAKE_C_COMPILER})
```

Comment:
> ${c_compiler_realpath}
> ${CMAKE_C_COMPILER}

Why just `c_compiler_realpath` doesn't work for you?
Expanding without quotes breaks on paths with spaces (most observed problem on Windows).

Variable expanding in not required with modern CMake ([2.8.12.2](https://cmake.org/cmake/help/v2.8.12/cmake.html#section_Policies)):
```
if(<variable|string> STREQUAL <variable|string>)
```

Reply:
Understood what you mean.
I just forgot to quote, and I didn't knew that I can avoid expanding.

Pushed without expanding.